Rogue-Ape-Studios/Crusader
# ✨ Conditional items framework

## Content

Created a rudimentary item system for the playtest.

## Changes

### Added
`Assets/Prefabs/Items/Basic Lifesteal Item Variant.prefab` : Was created, holds the Lifesteal on Hit item.
`Assets/Prefabs/Items/Crit Life Steal.prefab` : Was created, Holds the Lifesteal on Crit item.
`Assets/Prefabs/Items/Hitpoint Increase Variant.prefab` : Was created, Holds the MaxHP Increase item.
`Assets/Prefabs/Items/Last Stand Item Variant.prefab` : Was created, holds the Last Stand Item.
`Assets/Scripts/Items/CritLifeStealItem.cs` : Was created, and handles healing on a critical hit.
`Assets/Scripts/Items/LastStandItem.cs` : Was created, and handles the last stand mechanic.
`Assets/Scripts/Items/LifeStealItem.cs` : Was created, and handles healing on hit.

### Updated
`Assets/Prefabs/Items/ItemManager.prefab` : Was updated with the new items.
`Assets/Scripts/Health/PlayerHealth.cs` : Was updated with a revive mechanic.
`Assets/Scripts/Items/ItemManager.cs` : Was updated to handle the new items.

## Testing

1. Load the scene.
2. Open the item manager context menu.
3. Spawn an item.
4. Collect the item.
5. Observe Effects.


Closes #201
